nixos/modules/services/development/mlflow/default.nix

```diff
@@ -0,0 +1,137 @@
+{ options, config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.mlflow;
+
+  keyToFlag = k: (if lib.stringLength k == 1 then "-" else "--") + k;
+
+  toFlags = k: v:
+    if v == false || v == null then
+      ""
+    else if v == true then
+      keyToFlag k
+    else if lib.isList v then
+      lib.concatMapStringsSep " " (x: toFlags k x) v
+    else
+      "${keyToFlag k} ${toString v}";
+
+  flags = lib.concatStringsSep " " (lib.mapAttrsToList toFlags cfg.settings);
+
+  valueType = with lib.types;
+    nullOr (oneOf [ bool float int package path port str ]);
+
+  settingsType = with lib.types;
+    submodule {
+      freeformType = attrsOf (oneOf [ valueType (listOf valueType) ]);
```

Comment:
It is a complicated type that do allows `[ null null null ]` alas, wonder if it could be simpler even if you do accept more broken types and let runtime do its work, but no strong feeling.

Reply:
Oh, this is a bit of an experiment I guess. I was trying to come up with a generically usable flags type so you could specify flags as an attribute set. Most flags would take a single value, but I had to account for flags that might need to be specified repeatedly so I made it possible to use lists for that. The `nullOr` should probably be a layer higher so only one null can be specified. It's there to allow someone to override remove a flag with `mkForce null`.